walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -1,23 +1,189 @@
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//     https://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
+
 
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.*; 
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    //throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    List<TimeRange> possibleTimes = new ArrayList<TimeRange>();
+
+    //define variables
+    int dayStart = TimeRange.START_OF_DAY;
+    int dayEnd = TimeRange.END_OF_DAY;
+    int numberOfAttendees = request.getAttendees().size();
+    int numberOfEvents = events.size();
+    long meetingDuration = request.getDuration();
+
+    //optionsForNoAttendees
+    if(request.getAttendees().isEmpty())
+        return Arrays.asList(TimeRange.WHOLE_DAY);
+
+    //noOptionsForTooLongOfARequest
+    if(request.getDuration() > TimeRange.WHOLE_DAY.duration())
+        return Arrays.asList();
+    
+
+    //add attendees to attendees array
+    // String[] attendees = (String[])request.getAttendees().toArray();
+    List<String> attendees = new ArrayList<String>();
+    attendees.addAll(request.getAttendees());
+
+
+    //Times that don't work for all attendees
+    HashSet attendeeSchedule = new HashSet<TimeRange>();
+
+
+ 
+     
+
+    //find times attendees are busy
+    for(int i = 0 ; i < numberOfAttendees ; i++){
+        String currentAttendee = attendees.get(i);
+
+        for(Event event : events){
+            if(isAnAttendee(currentAttendee,event)){//checks if current attendant goes to this event
+                addToSchedule(attendeeSchedule, event);//adds event to current attendee schedule
+            }
+        }
+    }
+
+    
+    //ignores people not attending
+    if(attendeeSchedule.size()==0){
+        return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+
+        
+
+
+        List<TimeRange> busySchedule = new ArrayList<TimeRange>();
+            Iterator<TimeRange> myIt = attendeeSchedule.iterator(); 
+            while (myIt.hasNext()){
+                busySchedule.add(myIt.next());
+                
+            }
+
+        //sort times that don't work
+        Collections.sort(busySchedule, TimeRange.ORDER_BY_END);
+
+        //time nodes that are contained in others
+        trimContainedNodes(busySchedule);
+
+        
+
+        //test
```

Comment:
I would add some more comments to this `for` loops, since it looks like this is where a lot of your logic is happening. Maybe comment each `if` statement to explain what it's checking for?